### PR TITLE
Add subtitle in project description videos

### LINKED ISSUE #92: Play project video with subtitles (when provided)

### PR DESCRIPTION
**Pull request by PaulCoral**
Playing video with captions within the project description activity.
Subtitles are fetched from firebase.
Currently using the firebase cache.

This PR closes #92

### CODE REVIEW

**Comment by PaulCoral** [comment]
It is hard to get a better coverage as it only affect a few lines of code , checking subtitle relies on VideoView, and fetching them relies on Firebase-Firestore.
However, any suggestion is welcome

File: app/src/main/java/com/sdp13epfl2021/projmag/activities/ProjectInformationActivity.kt [hunk 1]
```diff
@@ -64,33 +67,61 @@ class ProjectInformationActivity : AppCompatActivity() {
...
+        val localUri = videosUris[current]
+        /* onlineUri is used as unique identifier for subtitles */
+        val onlineUri = projectVar.videoURI[current]
+        videoView.setVideoURI(localUri)
+        metadataDB.getSubtitlesFromVideo(
+            onlineUri,
+            Locale.ENGLISH.language,
+            { subs ->
+                subs?.let {
+                    videoView.addSubtitleSource(
```
Comment (arnauds5) [inline app/src/main/java/com/sdp13epfl2021/projmag/activities/ProjectInformationActivity.kt]: The subtitles are downloaded only when the video is played right ?
Maybe downloading once, when videos are downloaded could be better and could avoid to have no subtitle at the beginning if we have slow internet speed.

Comment (PaulCoral) [reply app/src/main/java/com/sdp13epfl2021/projmag/activities/ProjectInformationActivity.kt]: Yes good idea
It will be in the next commit

**Review by arnauds5 — COMMENTED** [review]
Tests are failling for some others tests (for example ListOfProjectsActivitiyTest), maybe the new popup for subtitle is shown during some tests, and the tests can't access the views.

**Review by arnauds5 — APPROVED** [review]
Looks good to me.
In https://github.com/asfoury/projmag/pull/123 I successfully run tests with fake databases and this increased the tests coverage.
It could be used for ProjectInformationActivity for example.

File: app/src/main/java/com/sdp13epfl2021/projmag/activities/ProjectInformationActivity.kt [hunk 2]
```diff
@@ -182,10 +192,17 @@ class ProjectInformationActivity : AppCompatActivity() {
     private fun addVideoAfterDownloaded(videosLinks: List<String>) {
         videosLinks.forEach { link ->
             fileDB.getFile(link, projectDir, { file ->
-                addVideo(Uri.fromFile(file))
-            }, {
-                showToast("An error occurred while downloading a video.")
-            })
```
Comment (arnauds5) [inline app/src/main/java/com/sdp13epfl2021/projmag/activities/ProjectInformationActivity.kt:188]: If the videos couldn't be downloaded, won't it be better to have a message to inform the user ?

**Comment by PaulCoral** [comment]
> Looks good to me.
> In #123 I successfully run tests with fake databases and this increased the tests coverage.
> It could be used for ProjectInformationActivity for example.

It could be possible, once we introduce dependency injection in activity (with Dagger Hilt), in the project.

File: app/src/main/java/com/sdp13epfl2021/projmag/activities/ProjectInformationActivity.kt (hunk 2 above)
Comment (PaulCoral) [reply app/src/main/java/com/sdp13epfl2021/projmag/activities/ProjectInformationActivity.kt:188]: I deleted it by accident
thank you

**Review by giogio21 — APPROVED** [review]

**Pull request merged**